client/utils/account-fees.tsx

```diff
@@ -74,28 +82,33 @@ const getFeeDescriptionString = ( fee ) => {
 	return '';
 };
 
-export const getCurrentFee = ( accountFees ) => {
+export const getCurrentFee = (
+	accountFees: FeeStructure
+): BaseFee | DiscountFee => {
 	return accountFees.discount.length
 		? accountFees.discount[ 0 ]
 		: accountFees.base;
 };
 
-export const formatMethodFeesTooltip = ( accountFees ) => {
+export const formatMethodFeesTooltip = (
+	accountFees: FeeStructure
+): JSX.Element | undefined => {
 	if ( ! accountFees ) return;
 
 	const total = {
 		percentage_rate:
 			accountFees.base.percentage_rate +
-			accountFees.additional.percentage_rate +
-			accountFees.fx.percentage_rate,
+			( accountFees.additional?.percentage_rate || 0 ) +
+			( accountFees.fx?.percentage_rate || 0 ),
```

Comment:
Looks fishy, are the definitions correct?

Reply:
thanks @kalessil  for checking it. As far as I understand the code, the `additional` and `fx` are optional properties so I marked them like that in the type definition. For that reason the TS compiler was complaining that those values could be `undefined`, that's way I fallback to 0.

I'll double check type definitions as you suggest.

Reply:
The fields are presented, but might have null as a value.

Speaking in definitions terms, that would be:
```php
export interface DiscountFee {
	end_time: string | null;
	volume_allowance: number | null;
	volume_currency: string | null;
	current_volume: number | null;
	percentage_rate: number;
	fixed_rate: number;
	discount: number|null;
}
```

To re-check the definition, please follow the trail in `\WCPay\Service\Fee_Service::get_account_fee_info` on the server side.

Reply:
Thanks @kalessil this is key point, I was not realizing that the properties were there but could have null values